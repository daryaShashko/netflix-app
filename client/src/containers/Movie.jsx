import React from 'react';

import Film from './../components/Film.jsx';
import FilmsAPI from './../FilmsAPI';
import Request from 'superagent';

export class Movie extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      movieId: '',
      movie: [],
    };

    this.getMovie = this.getMovie.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (parseInt(nextProps.movieId, 10) !== parseInt(this.state.movieId, 10)) {
      this.getMovie();
    }
  }

  componentDidMount() {
    this.setState({movieId: this.props.match.params.id});
  }

  getMovie() {
    let url = `https://api.themoviedb.org/3/movie/${this.props.match.params.id}?api_key=7ce8b734e18d23ac5345c37de7911c86`;
    Request.get(url).then((response) => {
      this.setState({
        movie: response.body,
      })
    })
  }

  render() {
    var movie = this.state.movie;
    return (
      <div className="grid__item grid__item_12" key={movie.id}>
        <Film
          poster={'https://image.tmdb.org/t/p/w500' + movie.poster_path}
          showTitle={movie.title}
          rating={movie.vote_average}
          releaseYear={movie.release_date}
          runtime={movie.runtime}
          summary={movie.overview}
        />
      </div>

    );
  }
}
