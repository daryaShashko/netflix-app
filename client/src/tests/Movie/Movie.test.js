/* eslint no-undef:0 */
import React from 'react';
import { configure, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Movie } from '../../containers/Movie.jsx';

import Adapter from 'enzyme-adapter-react-15';

configure({adapter: new Adapter()});


const setup = () => {
  const props = {
    movie: {
      id: 0,
      poster_path: '/path/to/poster1',
      title: 'Title1',
      vote_average: 4.7,
      release_date: '2007-02-03',
      runtime: 96,
      overview: 'Really great film!'
    }
  };

  const enzymeWrapper = mount(
    <Movie {...props} />
  );

  return {
    props,
    enzymeWrapper
  };
};

describe('Movie component', () => {
  const { enzymeWrapper } = setup();

  it('should render', () => {
    expect(toJson(enzymeWrapper)).toMatchSnapshot();
  });
});