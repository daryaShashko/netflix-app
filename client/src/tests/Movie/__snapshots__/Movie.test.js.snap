// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie component should render 1`] = `
<Movie
  movie={
    Object {
      "id": 0,
      "overview": "Really great film!",
      "poster_path": "/path/to/poster1",
      "release_date": "2007-02-03",
      "runtime": 96,
      "title": "Title1",
      "vote_average": 4.7,
    }
  }
>
  <div
    className="grid__item grid__item_12"
    key="0"
  >
    <Film
      category="Horror"
      director="director,"
      poster="https://image.tmdb.org/t/p/w500/path/to/poster1"
      rating={4.7}
      releaseYear="2007-02-03"
      runtime={96}
      showCast="showCast"
      showTitle="Title1"
      summary="Really great film!"
    >
      <div
        className="media-show"
      >
        <div
          className="media-show__wrapper"
        >
          <FilmPoster
            poster="https://image.tmdb.org/t/p/w500/path/to/poster1"
            showTitle="Batman"
            show_title="Title1"
          >
            <div
              className="media-show__poster"
            >
              <img
                alt="Batman"
                src="https://image.tmdb.org/t/p/w500/path/to/poster1"
              />
            </div>
          </FilmPoster>
          <div
            className="media-show__description"
          >
            <header
              className="media-show__header"
            >
              <FilmDescriptionItem
                className="media-show__title"
                text="Title1"
              >
                <span
                  className="media-show__title"
                >
                  Title1
                </span>
              </FilmDescriptionItem>
              <FilmDescriptionItem
                className="media-show__rating"
                text={4.7}
              >
                <span
                  className="media-show__rating"
                >
                  4.7
                </span>
              </FilmDescriptionItem>
              <FilmDescriptionItem
                className="media-show__category"
                text="Horror"
              >
                <span
                  className="media-show__category"
                >
                  Horror
                </span>
              </FilmDescriptionItem>
              <FilmDescriptionItem
                className="media-show__release-year"
                text="2007-02-03"
              >
                <span
                  className="media-show__release-year"
                >
                  2007-02-03
                </span>
              </FilmDescriptionItem>
              <FilmDescriptionItem
                className="media-show__runtime"
                text={96}
              >
                <span
                  className="media-show__runtime"
                >
                  96
                </span>
              </FilmDescriptionItem>
            </header>
            <FilmDescriptionItem
              className="media-show__summary"
              text="Really great film!"
            >
              <span
                className="media-show__summary"
              >
                Really great film!
              </span>
            </FilmDescriptionItem>
            <footer
              className="media-show__footer"
            >
              <FilmDescriptionItem
                className="media-show__director"
                text="Director: director,"
              >
                <span
                  className="media-show__director"
                >
                  Director: director,
                </span>
              </FilmDescriptionItem>
              <FilmDescriptionItem
                className="media-show__cast"
                text="Cast: showCast"
              >
                <span
                  className="media-show__cast"
                >
                  Cast: showCast
                </span>
              </FilmDescriptionItem>
            </footer>
          </div>
        </div>
      </div>
    </Film>
  </div>
</Movie>
`;
